pkg/telemetry/utils.go

```diff
@@ -0,0 +1,59 @@
+package telemetry
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/warpbuilds/warpbuild-agent/pkg/warpbuild"
+)
+
+func readLastNLines(filePath string, n int) ([]byte, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file: %w", err)
+	}
+	defer file.Close()
+	var lines []string
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
```

Comment:
memory issue can be here.

Reply:
On second thought, memory issue won't be here, since we are only keeping n lines in the array. But new array is being created and removed for each line you can optimize that if you wish.

Reply:
Yeah, the array only keeps n lines in the memory at a time. Reading a file backwards requires some complicated logic of seeking etc. which I don't think is required.